src/blocks/gallery-carousel/edit.js

```diff
@@ -66,16 +65,6 @@ const GalleryCarouselEdit = ( props ) => {
 		}
 	}, [ props.attributes.gutter ] );
 
-	useEffect( () => {
-		if (
-			props.attributes.gridSize === 'xlrg' &&
-			prevAlign === undefined &&
-			( gutter !== 0 || gutterMobile !== 0 )
-		) {
-			props.setAttributes( { gutter: 0, gutterMobile: 0 } );
-		}
-	}, [ props.attributes.gridSize, prevAlign ] );
-
```

Comment:
I believe the original intention here was to prevent a gutter value from setting while the block has xlrg grid size. That logic does not seem to be working as expected and a few UX concerns. If we intend to prevent users from using a gutter while the block is set to xlrg size then we should hide the controls altogether as well as include this logic. 

Actually diving in a bit further I see that the prevAlign may not be calculating correctly or something like that because when the alignment is `unset` the controls are indeed hidden. (Master)
![AlignSizeControlsHidden](https://user-images.githubusercontent.com/30462574/132565721-0c8e5b5a-75c1-4592-84d2-47f2e40c4049.gif)


This functionality should persist in this PR.

Reply:
We Might be able to give this a quick fix by basing the logic on not what the previous value is but what the current calculated value is. The same way in which we hide the controls.